### PR TITLE
Merge devel

### PR DESCRIPTION
**Pull request by lubo**

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -1,4 +1,4 @@
 cdist manifestation
 ===================
 
-A library which helps you with calling cdist types from manifests written in Python.
+A library for calling cdist types from manifests written in Python.
```
Comment (adam-dej) [inline README.rst]: What about "A helper for writing cdist manifests in Python"?

File: cdist_manifestation/dependencies.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
+from contextlib import contextmanager
+
+_dependencies = []
+
+
+@contextmanager
+def require(*args):
```
Comment (adam-dej) [inline cdist_manifestation/dependencies.py]: Shouldn't we support also an equivalent of setting `CDIST_ORDER_DEPENDENCY` variable?

File: README.rst (hunk 1 above)
Comment (lubo) [reply README.rst]: That's better, changed.

File: cdist_manifestation/dependencies.py (hunk 2 above)
Comment (lubo) [reply cdist_manifestation/dependencies.py]: We should. `order_dependency()` added.

File: cdist_manifestation/dependencies.py [hunk 3]
```diff
@@ -1,6 +1,16 @@
 from contextlib import contextmanager
 
 _dependencies = []
+_order_dependency = False
+
+
+def order_dependency(value):
```
Comment (adam-dej) [inline cdist_manifestation/dependencies.py]: Can't we make this `contextmanager` as well?

```python
with order_dependency():
    typeA()
    typeB()

typeC()
```

Comment (lubo) [reply cdist_manifestation/dependencies.py]: Yeah, I like that better. Changed now.

File: cdist_manifestation/dependencies.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
+from contextlib import contextmanager
+
+_dependencies = []
+_order_dependency = False
+
+
+@contextmanager
+def order_dependency():
```
Comment (adam-dej) [inline cdist_manifestation/dependencies.py:8]: What happens if somebody nests `order_dependency()`?

File: example/conf/manifest/init [hunk 5]
```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env python3
+
+from cdist_manifestation.dependencies import order_dependency, require
+from cdist_manifestation.types import file, directory
+
+with order_dependency():
```
Comment (adam-dej) [inline example/conf/manifest/init]: This example does not make much sense since only one object is created here

**Review by adam-dej — CHANGES_REQUESTED** [review]

File: example/conf/manifest/init (hunk 5 above)
Comment (lubo) [reply example/conf/manifest/init]: Fixed.

File: cdist_manifestation/dependencies.py (hunk 4 above)
Comment (lubo) [reply cdist_manifestation/dependencies.py:8]: Fixed.

File: example/conf/manifest/init (hunk 5 above)
Comment (lubo) [reply example/conf/manifest/init]: Example changed to be more sensible.

**Review by adam-dej — APPROVED** [review]

**Pull request merged**